troubleshooting/known_issues/ims_remote_node_image_build_failure.md

```diff
@@ -0,0 +1,245 @@
+    +- name: csm-noos
+    +  description: "CSM No-OS Packages (added by Ansible)"
+    +  repo: https://packages.local/repository/csm-noos
+    +- name: csm-sle
+    +  description: "CSM SLE Packages (added by Ansible)"
+    +  repo: 'https://packages.local/repository/csm-sle-${releasever_major}sp${releasever_minor}'
+    +- name: csm-noos
+    +  description: "CSM No-OS Packages (added by Ansible)"
+    +  repo: https://packages.local/repository/csm-noos
+    +- name: csm-embedded
+    +  description: "CSM Embedded NCN Packages (added by Ansible)"
+    +  repo: https://packages.local/repository/csm-embedded
+    
+    diff --git a/vars/csm_packages.yml b/vars/csm_packages.yml
+    index 8c9c67e..8fb8fe2 100644
+    --- a/vars/csm_packages.yml
+    +++ b/vars/csm_packages.yml
+    @@ -77,6 +77,14 @@ compute_csm_sles_packages:
+       - cray-uai-util
+       - cray-spire-dracut>=2.0.0
+    
+    -# IMS Remote Compute Nodes:
+    -ims_compute_sles_packages:
+    -  - podman
+    +ims_csm_sles_packages:
+    +  - cfs-debugger
+    +  - cfs-state-reporter
+    +  - cfs-trust
+    +  - craycli
+    +  - csm-auth-utils
+    +  - csm-node-heartbeat
+    +  - csm-node-identity
+    +  - spire-agent>=1.5.0
+    +  - tpm-provisioner-client
+    +  - cni-plugins=1.1.1-150500.3.2.1
+    ```
+
+1. (`ncn-mw#`) Check out the `csm-config` Ansible plays
+
+    The Ansible plays for configuring images are stored a git repository on the system. To check out
+    the repository:
+
+    ```bash
+    VCS_USER=$(kubectl get secret -n services vcs-user-credentials --template={{.data.vcs_username}} | base64 --decode)
+    VCS_PASSWORD=$(kubectl get secret -n services vcs-user-credentials --template={{.data.vcs_password}} | base64 --decode)
+    git clone https://$VCS_USER:$VCS_PASSWORD@api-gw-service-nmn.local/vcs/cray/csm-config-management.git
+    ```
+
+    Expected output will look something like:
+
+    ```text
+    Cloning into 'csm-config-management'...
+    remote: Enumerating objects: 262, done.
+    remote: Counting objects: 100% (262/262), done.
+    remote: Compressing objects: 100% (96/96), done.
+    remote: Total 262 (delta 90), reused 259 (delta 90), pack-reused 0
+    Receiving objects: 100% (262/262), 64.19 KiB | 10.70 MiB/s, done.
+    Resolving deltas: 100% (90/90), done.
+    ```
+
+    Now set up a temporary branch based on the branch for the installed version os CSM:
+
+    ```bash
+    cd csm-config-management
+    git checkout $CSM_IMPORT_BRANCH
+    git checkout -b patch_branch
+    ```
+
+    Expected output will look something like:
+
+    ```text
+    # git checkout cray/csm/1.27.2
+    branch 'cray/csm/1.27.2' set up to track 'origin/cray/csm/1.27.2'.
+    Switched to a new branch 'cray/csm/1.27.2'
+    # git checkout -b patch_branch
+    Switched to a new branch 'patch_branch'
+    ```
+
+1. (`ncn-mw#`) Apply a fix via a git patch operation
+
+    ```bash
+    git apply ../fix.patch
+    git commit -a -m "Apply WAR patch."
```

Comment:
This could be a more specific commit message like 'Fixed bare bones CFS error with missing package <blah>'.

Reply:
Done.